core/model/modx/processors/search/search.class.php

```diff
@@ -18,40 +33,584 @@ class modSearchProcessor extends modProcessor
     public function process()
     {
         $this->query = $this->getProperty('query');
+
         if (!empty($this->query)) {
-            if (strpos($this->query, ':') === 0) {
-                // upcoming "launch actions"
-                //$this->searchActions();
+            if (strpos($this->query, ':') !== 0) {
+                // By default only search in resources
+                $this->query = ':r:*:' . $this->query;
+            }
+            $this->query = explode(':', $this->query);
+            $this->doSearch();
+        }
+
+        return $this->outputArray($this->results);
+    }
+
+    public function doSearch()
+    {
+        $query = $this->query;
+        /* What are we looking for? */
+        $element = isset($query[1]) ? $query[1] : false;
+
+        $count = count($query);
+        $result['count'] = $count;
+
+        $btn = "x-btn x-btn-small x-btn-icon-small-left primary-button x-btn-noicon";
+
+        $result['uberbar_header'] = $this->modx->lexicon('uberbar_header');
+
+        /* : */
+        if ($count == 2) {
+            $validate = $this->validateUberbar($query);
+
+            if ($validate === true) {
+                // Return list of available resource types
+                $uberbar_mode = $this->modx->getOption('uberbar_mode');
+                $result['header'] = $this->modx->lexicon('uberbar_pick_element_header');
+                $result['msg'] = $this->modx->lexicon('uberbar_pick_element_msg', array('uberbar_mode' => $uberbar_mode));
+                $options = '';
+                foreach ($this->elements as $key => $val) {
+                    /* Only show elements you have permission to see. */
+                    if ($val['permission'] === true) {
+                        $activeClass = ($key == $query[1]) ? 'active ' : '';
+                        $options .= '<li><a data-value="' . $key . '" class="' . $activeClass . $btn . '">' . $val['label'] . '</a></li>';
+                    }
+                }
+                $result['options'] = $options;
+
+                $this->results[] = $result;
```

Comment:
I guess we could stop the execution once the result has been added (no chance `$count` will be different than `2` here, right ? ;))